sdk/storage/Azure.Storage.DataMovement/src/TransferItemScheduler.cs

```diff
@@ -0,0 +1,156 @@
+
+        /// <summary>
+        /// The size of subsequent ranges.
+        /// </summary>
+        private readonly long _rangeSize;
+
+        /// <summary>
+        /// The size we use to determine whether to upload as a one-off request or
+        /// a partitioned/committed upload
+        /// </summary>
+        private readonly long _singleUploadThreshold;
+
+        // TaskScheduler
+        //
+        // To manage something like this that will eventually exit when the queue empties
+        // We might have to consider making a custom TaskScheduler to start a new one, once
+        // the queue has something to process.
+        //private TaskScheduler taskScheduler;
+
+        /// <summary>
+        /// The queue we hold onto as the finished scanned items are added. This is the
+        /// waiting list of items ready to execute (upload, download, copy)
+        /// </summary>
+        public AsyncQueue<StorageTransferJob> itemsToProcess;
+
+        /// <summary>
+        /// TransferItemScheduler Constructor
+        /// </summary>
+        /// <param name="transferOptions"></param>
+        public TransferItemScheduler(StorageTransferOptions transferOptions = default)
+        {
+            // Set _maxWorkerCount
+            if (transferOptions.MaximumConcurrency.HasValue
+                && transferOptions.MaximumConcurrency > 0)
+            {
+                _maxWorkerCount = transferOptions.MaximumConcurrency.Value;
+            }
+            else
+            {
+                _maxWorkerCount = Constants.Blob.Block.DefaultConcurrentTransfersCount;
+            }
+
+            // Set _rangeSize
+            if (transferOptions.MaximumTransferSize.HasValue
+                && transferOptions.MaximumTransferSize.Value > 0)
+            {
+                _rangeSize = Math.Min(transferOptions.MaximumTransferSize.Value, Constants.Blob.Block.MaxDownloadBytes);
+            }
+            else
+            {
+                _rangeSize = Constants.DefaultBufferSize;
+            }
+
+            // Set _initialRangeSize
+            if (transferOptions.InitialTransferSize.HasValue
+                && transferOptions.InitialTransferSize.Value > 0)
+            {
+                _initialDownloadRangeSize = transferOptions.InitialTransferSize.Value;
+            }
+            else
+            {
+                _initialDownloadRangeSize = Constants.Blob.Block.DefaultInitalDownloadRangeSize;
+            }
+
+            // Set _singleUploadThreshold
+            if (transferOptions.InitialTransferSize.HasValue
+                && transferOptions.InitialTransferSize.Value > 0)
+            {
+                _singleUploadThreshold = Math.Min(transferOptions.InitialTransferSize.Value, Constants.Blob.Block.MaxUploadBytes);
+            }
+            else
+            {
+                _singleUploadThreshold = Constants.Blob.Block.Pre_2019_12_12_MaxUploadBytes;
+            }
```

Comment:
I don't see these being used. 
Assuming we're going to use components from blobs to do actual transfers then they should already have these defaults built in. I'd expect DM lib to provide overrides for defaults, so not sure why do we have this.

Reply:
Removed.